napari/plugins/io.py

```diff
@@ -278,6 +316,12 @@ def _write_single_layer_with_plugins(
         plugin_manager.hook, f'napari_write_{layer._type_string}'
     )
 
+    if plugin_name and (plugin_name not in plugin_manager.plugins):
+        raise ValueError(
+            f"There is no registered plugin named {plugin_name!r}.\n"
+            f"Plugin names include: {set(plugin_manager.plugins)!r}"
```

Comment:
Why is this segment repeated? I can’t follow the flow...

Reply:
one is for readers (coming from `viewer.open`) the other two are for writers (coming from `save_layers`)...  In both cases, I put the check as low as possible in the call chain, so it can't be missed.  For writers, I _could_ move the ones in `_write_multiple_layers_with_plugins` and `_write_single_layer_with_plugins` up into `save_layers` (and let it be a more cryptic error if someone calls the private functions incorrectly) but then it would also be best if I added a `plugin_manager` argument to that public function, which I was trying to avoid.  let me know what you prefer

Reply:
perhaps also worth pointing out... this check is technically not necessary at all.  It would still be caught in the hookcaller, but in that case the error message would be something like `"No implementation of napari_get_reader found for plugin ___"` ... which makes it sound like there _might_ be a plugin with that name... (but the hookcaller doesn't have access to the plugin manager and doesn't know that for sure)